src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -50,10 +50,10 @@ protected override void InitializeCompareExchangeStorage()
         CompareExchangeStorage.Initialize(ShardedDatabaseName);
     }
 
-    protected override void InitializeAndStartPeriodicDocumentsMigrator()
+    protected override void InitializeAndStartDocumentsMigration()
     {
-        PeriodicDocumentsMigrator = new ShardedPeriodicDocumentsMigrator(this);
-        PeriodicDocumentsMigrator.Start();
+        DocumentsMigrator = new ShardedDocumentsMigrator(this);
+        Task.Run(DocumentsMigrator.ExecuteMoveDocumentsAsync);
```

Comment:
At this point the database is accessible and initialized?

Reply:
Done

Reply:
Why `Task.Run`?